File: count_total_set_bits.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
 public:
  // n: input to count the number of set bits
  // Function to return sum of count of set bits in the integers from 1 to n.
  int countSetBits(int n) {
    // Your logic here
    unordered_map<int, int> m;
    m[0] = 0;
    m[1] = 1;
    m[2] = 2;
    int k = 2;
    for (int i = 4; i <= n; i *= 2) {
      m[i] = k * (i / 2) + 1;
      ++k;
    }
    int i = 1;
    for (; i * 2 <= n; i *= 2)
      ;
    // cout << "i = " << i << endl;
    int ans = 0;
    for (int j = i; j > 0; j /= 2) {
      // cout << j << endl;
      if (j <= n) {
        ans += m[j];
        n -= j;
        ans += n;
      }
    }

    return ans;
  }
};

//{ Driver Code Starts.

// Driver code
int main() {
  int t;
  cin >> t;    // input testcases
  while (t--)  // while testcases exist
  {
    int n;
    cin >> n;  // input n
    Solution ob;
    cout << ob.countSetBits(n) << endl;  // print the answer
  }
  return 0;
}

// } Driver Code Ends

File: bit/count_total_setbits.py
# User function Template for python3


class Solution:
    def countBits(self, N):
        # code here
        ans = 0
        for i in range(31, 0, -1):
            if N & (1 << i):
                ans += N % (1 << i) + 1 + (1 << i - 1) * i
                N &= (1 << i) - 1
        if N % 2:
            ans += 1

        return ans


# {
# Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        print(ob.countBits(N))
# } Driver Code Ends
